Test `reposerver/repository/repository_test/TestInit` broken when using Docker Desktop on MacOS

Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

The test `reposerver/repository/repository_test/TestInit()` fails 
with the containerized toolchain when using Docker Desktop on Macos.

<!-- A clear and concise description of what the bug is. -->

**To Reproduce**

Tested on recent macos (arm64 Sequoia 15.5).

1. clone Argo-Cd from its master branch, cd into the project directory
2. Start Docker Desktop
3. run `make TEST_MODULE=github.com/argoproj/argo-cd/v3/reposerver/repository test`
4. See the test fail with the message:

=== RUN   TestInit
    repository_test.go:3121: 
        	Error Trace:	/go/src/github.com/argoproj/argo-cd/reposerver/repository/repository_test.go:3121
        	Error:      	An error is expected but got nil.
        	Test:       	TestInit
--- FAIL: TestInit (0.10s)

**Expected behavior**

The test should pass.

**Version**

```shell
argocd: v3.2.0+4024fe7
  BuildDate: 2025-08-03T07:12:06Z
  GitCommit: 4024fe7c22db2f8a575a534be3295e21df2c2d23
  GitTreeState: clean
  GoVersion: go1.24.4
  Compiler: gc
  Platform: darwin/arm64
```

**Logs**

```
$ make TEST_MODULE=github.com/argoproj/argo-cd/v3/reposerver/repository test
docker build --build-arg UID=501 -t argocd-test-tools -f test/container/Dockerfile .

[SKIP build output]

docker tag argocd-test-tools argocd-test-tools:latest
mkdir -p /Users/dudin/.cache/go-build
docker run --rm -it --name argocd-test-client -u 501:20 -e HOME=/home/user -e GOPATH=/go -e ARGOCD_E2E_K3S= -e GITHUB_TOKEN -e GOCACHE=/tmp/go-build-cache -e ARGOCD_LINT_GOGC=20 -v "/Users/dudin/go/src/github.com/argoproj/argo-cd-MY:/go/src/github.com/argoproj/argo-cd:delegated" -v /Users/dudin/go/pkg/mod:/go/pkg/mod:delegated -v /Users/dudin/.cache/go-build:/tmp/go-build-cache:delegated -v /Users/dudin/.kube:/home/user/.kube:delegated -v /tmp:/tmp:delegated -w /go/src/github.com/argoproj/argo-cd  argocd-test-tools:latest bash -c "make TEST_MODULE=github.com/argoproj/argo-cd/v3/reposerver/repository test-local"
if test "github.com/argoproj/argo-cd/v3/reposerver/repository" = ""; then \
	DIST_DIR=/go/src/github.com/argoproj/argo-cd/dist RERUN_FAILS=0 PACKAGES=`go list ./... | grep -v 'test/e2e'` ./hack/test.sh -args -test.gocoverdir="/go/src/github.com/argoproj/argo-cd/test-results"; \
else \
	DIST_DIR=/go/src/github.com/argoproj/argo-cd/dist RERUN_FAILS=0 PACKAGES="github.com/argoproj/argo-cd/v3/reposerver/repository" ./hack/test.sh -args -test.gocoverdir="/go/src/github.com/argoproj/argo-cd/test-results" "github.com/argoproj/argo-cd/v3/reposerver/repository"; \
fi
+ which go-junit-report
/go/bin/go-junit-report
+ TEST_RESULTS=test-results
+ TEST_FLAGS=
+ test '' '!=' ''
+ test '' '!=' ''
+ mkdir -p test-results
+ GODEBUG=tarinsecurepath=0,zipinsecurepath=0
+ gotestsum --rerun-fails-report=rerunreport.txt --junitfile=test-results/junit.xml --format=testname --rerun-fails=0 --packages=github.com/argoproj/argo-cd/v3/reposerver/repository -- -cover -args -test.gocoverdir=/go/src/github.com/argoproj/argo-cd/test-results github.com/argoproj/argo-cd/v3/reposerver/repository

[SKIP passed tests]

=== RUN   TestInit
    repository_test.go:3121: 
        	Error Trace:	/go/src/github.com/argoproj/argo-cd/reposerver/repository/repository_test.go:3121
        	Error:      	An error is expected but got nil.
        	Test:       	TestInit
--- FAIL: TestInit (0.10s)
FAIL reposerver/repository.TestInit (0.10s)

[SKIP]

DONE 245 tests, 1 failure in 40.275s
make: *** [Makefile:415: test-local] Error 1
make: *** [test] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -138,7 +138,6 @@ define run-in-test-client
 		-v ${GOPATH}/pkg/mod:/go/pkg/mod${VOLUME_MOUNT} \
 		-v ${GOCACHE}:/tmp/go-build-cache${VOLUME_MOUNT} \
 		-v ${HOME}/.kube:/home/user/.kube${VOLUME_MOUNT} \
-		-v /tmp:/tmp${VOLUME_MOUNT} \
 		-w ${DOCKER_WORKDIR} \
 		$(PODMAN_ARGS) \
 		$(TEST_TOOLS_PREFIX)$(TEST_TOOLS_IMAGE):$(TEST_TOOLS_TAG) \

--- a/Makefile
+++ b/Makefile
@@ -113,7 +113,6 @@ define run-in-test-server
 		-v ${GOPATH}/pkg/mod:/go/pkg/mod${VOLUME_MOUNT} \
 		-v ${GOCACHE}:/tmp/go-build-cache${VOLUME_MOUNT} \
 		-v ${HOME}/.kube:/home/user/.kube${VOLUME_MOUNT} \
-		-v /tmp:/tmp${VOLUME_MOUNT} \
 		-w ${DOCKER_WORKDIR} \
 		-p ${ARGOCD_E2E_APISERVER_PORT}:8080 \
 		-p 4000:4000 \

--- a/docs/developer-guide/toolchain-guide.md
+++ b/docs/developer-guide/toolchain-guide.md
@@ -23,7 +23,6 @@ The Docker container for the virtualized toolchain will use the following local 
 * `~/go/src` - Your Go workspace's source directory (modifications expected)
 * `~/.cache/go-build` - Your Go build cache (modifications expected)
 * `~/.kube` - Your Kubernetes client configuration (no modifications)
-* `/tmp` - Your system's temp directory (modifications expected)
 
 #### Known issues on macOS
 [Known issues](mac-users.md)

